api/src/test/java/org/openmrs/util/ConceptReferenceRangeUtilityTest.java

```diff
@@ -0,0 +1,292 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util;
+
+import org.apache.velocity.exception.ParseErrorException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.openmrs.Concept;
+import org.openmrs.ConceptDatatype;
+import org.openmrs.Encounter;
+import org.openmrs.Location;
+import org.openmrs.Obs;
+import org.openmrs.Person;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.ValidationException;
+import org.openmrs.test.jupiter.BaseContextSensitiveTest;
+
+import java.time.LocalTime;
+import java.util.Calendar;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+class ConceptReferenceRangeUtilityTest extends BaseContextSensitiveTest {
+	private Calendar calendar;
+	private Person person;
+	
+	@Mock
+	private ObsService obsService;
+	
+	@BeforeEach
+	public void setUp() {
+		calendar = Calendar.getInstance();
+		person = new Person();
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnTrueIfAgeIsWithinRange() {
+		calendar.add(Calendar.YEAR, -5);
+		person.setBirthdate(calendar.getTime());
+		String criteria = "$patient.getAge() > 1 && $patient.getAge() < 10";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnFalseIfAgeIsOutsideRange() {
+		calendar.add(Calendar.YEAR, -11);
+		person.setBirthdate(calendar.getTime());
+		String criteria = "$patient.getAge() > 1 && $patient.getAge() < 10";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnTrueIfAgeIsOnBoundary() {
+		calendar.add(Calendar.YEAR, -10);
+		person.setBirthdate(calendar.getTime());
+		String criteria = "$patient.getAge() >= 1 && $patient.getAge() <= 10";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnFalseIfAgeIsNoMatch() {
+		calendar.add(Calendar.YEAR, -2);
+		person.setBirthdate(calendar.getTime());
+		String criteria = "$patient.getAge() >= 15 && $patient.getAge() <= 50";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInYears_shouldReturnTrueIfAgeIsWithinRange() {
+		Person person = new Person();
+
+		// Set birthdate to 5 years ago from today
+		Calendar calendar = Calendar.getInstance();
+		calendar.add(Calendar.YEAR, -5);
+		Date birthDate = calendar.getTime();
+		person.setBirthdate(birthDate);
+
+		String criteria = "$fn.getAge($patient, 'YEARS') > 1 && $fn.getAge($patient, 'YEARS') < 10";
```

Comment:
Why would one use `$fn.getAge($patient, 'YEARS')` instead of $patient.getAge()`?

Reply:
Let us test with criteria that we intend to be used. It will be way much simpler for one to do $fn.getAgeInMonths() and $fn.getAgeInWeeks() or $fn.getAgeInDays() than any other more complicated syntax. Therefore, since patient.getAge() is already returning in years, let us just add three utility methods to return in days, weeks, and months.